src/SeleniumLibrary/keywords/element.py

```diff
@@ -112,21 +113,42 @@ def page_should_contain_element(self, locator, message=None, loglevel='INFO'):
         The ``message`` argument can be used to override the default error
         message.
 
+        The ``limit`` argument can used to define how many elements the
+        page should contain. When ``limit`` is `None` page can contain one or
+        more elements. When limit is a number, page must contain same number
+        of elements.
+
         See `Page Should Contain` for explanation about the ``loglevel``
         argument.
+
+        The ``limit`` argument is new in SeleniumLibrary 3.0
+
+        Examples assumes that locator matches to two elements.
+        | Page Should Contain Element | name: div_name | limit=1    | # Keyword fails.                  |
+        | Page Should Contain Element | name: div_name | limit=2    | # Keyword passes.                 |
+        | Page Should Contain Element | name: div_name | limit=none | # None is considered one or more. |
+        | Page Should Contain Element | name: div_name |            | # Same as above.                  |
         """
-        self.assert_page_contains(locator, message=message, loglevel=loglevel)
+        if is_noney(limit):
+            self.assert_page_contains(locator, message=message,
+                                      loglevel=loglevel)
+        else:
+            limit = int(limit)
+            count = len(self.find_elements(locator))
+            if count == limit:
+                self.info('Current page contains {} element(s).'
+                          .format(count))
+            else:
+                if is_noney(message):
+                    message = ('Page should have contained "{}" element(s), '
+                               'but it did contain "{}" element(s).'
+                               .format(limit, count))
+                self.ctx.log_source(loglevel)
+                raise AssertionError(message)
 
     @keyword
     def locator_should_match_x_times(self, locator, x, message=None, loglevel='INFO'):
-        """Verifies that ``locator`` matches ``x`` number of elements.
-
-        See the `Locating elements` section for details about the locator
-        syntax.
-
-        See `Page Should Contain Element` for explanation about ``message``
-        and ``loglevel`` arguments.
-        """
+        """Deprecated, use `Page Should Contain Element` with `limit` argument instead."""
```

Comment:
Should have double backticks around `limit` (and Libdoc should be enhanced to warn about using single backticks when not linking but that would be a backwards incompatible change).